src/main/java/teammates/ui/webapi/output/StudentData.java

```diff
@@ -89,6 +106,16 @@ public void setJoinState(JoinState joinState) {
         this.joinState = joinState;
     }
 
+    public void setKey(List<StudentAttributes> studentAttributes) {
+        this.key = StringHelper.encrypt(studentAttributes.stream()
+                .filter((StudentAttributes s) -> s.getGoogleId().equals(this.googleId))
```

Comment:
I think we might want to improve this?

Iterating the whole array to find the corresponding element is not really attracting.

Reply:
Hey Xiao Pu,
I think there were two alternatives I considered when doing this.
First, is to put the key generation inside the constructor. However, this will place the burden of hiding the `key` on the developer. Since Teammates has many contributors, I opted not to do this as people who use StudentsData in the future may not be aware of this and thus send the key.

Yea, theres a runtime issue, but I think this should be fine as its Java, so it should run faster and because the performance bottleneck is at the frontend side.

What do you think?

Reply:
> since Teammates has many contributors, I opted not to do this as people who use StudentsData in the future may not be aware of this and thus send the key.

I don't agree with this as RESTfuL endpoints are enforced by resources. `Student` is a resource and the only two places till now is search for the resource or get the resource directly.

Despite this, there is a still a way to do this
```java
studentsData = new StudentsData();
List<StudentData> studentDataList = new ArrayList<>();
for (StudentAttributes student : studentAttributesList) {
    // call constructor
    // generate key depends on access control
}
studentsData.setStudents(studentDataList)
```

Reply:
Hey Xiao Pu, that is a fair point about the two places thing. Will change 👍